overlord/devicestate/devicestate.go

```diff
@@ -116,6 +116,23 @@ func findSerial(st *state.State, device *auth.DeviceState) (*asserts.Serial, err
 	return a.(*asserts.Serial), nil
 }
 
+// findRevisionForModel returns the model assertion revision if any in the
+// assertion database for the given model, otherwise it returns -1.
+func findRevisionForModel(st *state.State, mod *asserts.Model) (modRevision int, err error) {
+	a, err := assertstate.DB(st).Find(asserts.ModelType, map[string]string{
+		"series":   release.Series,
+		"brand-id": mod.BrandID(),
+		"model":    mod.Model(),
+	})
+	if errors.Is(err, &asserts.NotFoundError{}) {
```

Comment:
apologies if I'm not up to date with the changes, but won't this break a remodel of say ubuntu-core-20 to ubuntu-core-22 if the model wasn't acked before?

Reply:
if the new model is not there we return -1 and no error, so  new.Revision() < -1 will be false, which should be ok, no?

Reply:
ah sorry, my bad!